pkg/sloop/webserver/debug.go

```diff
@@ -162,6 +164,107 @@ func listKeysHandler(tables typed.Tables) http.HandlerFunc {
 	}
 }
 
+type SloopKeyInfo struct {
+	TotalKeys     int64
+	EstimatedSize int64
+}
+
+type SloopKey struct {
+	TableName   string
+	PartitionID string
+}
+
+type histogram struct {
+	HistogramMap map[SloopKey]*SloopKeyInfo
+	TotalKeys    int
+	DeletedKeys  int
+}
+
+// returns IsDeletedExpired, TableName, PartitionId, EstimatedSize, error.
+func ParseSloopKey(item badgerwrap.Item) (bool, string, string, int64, error) {
+	key := item.Key()
+	parts := strings.Split(string(key), "/")
+	if len(parts) != 7 {
+		return false, "", "", 0, fmt.Errorf("key should have 6 parts: %x", key)
+	}
+
+	if parts[0] != "" {
+		return false, "", "", 0, fmt.Errorf("key should start with /: %x", key)
+	}
+
+	var isDeletedExpired = item.IsDeletedOrExpired()
+	var tableName = parts[1]
+	var partitionId = parts[2]
+	return isDeletedExpired, tableName, partitionId, item.EstimatedSize(), nil
```

Comment:
mixed paradigms - consider moving `item.EstimatedSize()` to be in a local variable like the others